### ISSUE TITLE
Deprecate all Linux kernel APIs

### ISSUE THREAD

**Issue by gnzlbg** [rust-lang/libc#1391]
Libc currently only exposes a lot of Linux kernel APIs, but leaves doing so to other crates for other operating systems. 

We probably should do the same that we do for MacOS and Windows, that is, have a different crate, e.g., `linux`, that's just like `mach`, `winapi`, etc. and that it is tested against different Linux kernel versions.

Once we have that in place, we can start deprecating kernel APIs from here, and recommend users to use other crates instead.

**Comment by newpavlov** [comment]
Good idea! Unfortunately `linux` crate is owned by @mahkoh and before doing deprecation I would prefer for it to be transferred to someone more active and interested in maintaining it.

**Comment by mahkoh** [comment]
Nope.

**Comment by gnzlbg** [comment]
@newpavlov I have a `linux-rs` crate that uses rust-bindgen to auto generate the headers.
